lib/api/src/native.rs

```diff
@@ -112,8 +112,26 @@ macro_rules! impl_native_traits {
             $( $x: FromToNativeWasmType, )*
             Rets: WasmTypeList,
         {
+
             /// Call the typed func and return results.
             pub fn call(&self, $( $x: $x, )* ) -> Result<Rets, RuntimeError> {
+                self.call_(true, $( $x, )*)
+            }
+
+            /// Call the typed func and return results.
+            /// This would not catch any traps and must be called within a callback passed to `wasmer_vm::catch_traps`.
+            pub fn call_unchecked(&self, $( $x: $x, )* ) -> Result<Rets, RuntimeError> {
+                self.call_(false, $( $x, )*)
+            }
+
+            /// Get a VM Context to manually handle traps when calling `call_unchecked`.
+            pub fn get_vmctx(&self) -> VMFunctionEnvironment {
+                self.exported.vm_function.vmctx
+            }
+
+            /// Call the typed func and return results.
+            /// When calling a wasm function, allows to do checked or unchecked call.
+            fn call_(&self, trampoline_checked: bool, $( $x: $x, )* ) -> Result<Rets, RuntimeError> {
```

Comment:
This function must be `unsafe` and have the prerequisites for correct usage documented with a `# Safety` section.

Also, we can avoid the runtime check here as well.